pkg/asset/installconfig/openstack/validation/platform.go

```diff
@@ -37,26 +40,78 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, ci *CloudInfo)
 	return allErrs
 }
 
-// validateMachinesSubnet validates the machines subnet and enforces proper byo subnet usage and returns a list of all validation errors
-func validateMachinesSubnet(p *openstack.Platform, n *types.Networking, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
-	if p.MachinesSubnet != "" {
-		if len(p.ExternalDNS) > 0 {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("externalDNS"), p.ExternalDNS, "externalDNS is set, externalDNS is not supported when machinesSubnet is set"))
-		}
-		if ci.MachinesSubnet == nil {
-			allErrs = append(allErrs, field.NotFound(fldPath.Child("machinesSubnet"), p.MachinesSubnet))
-		} else if !validUUIDv4(p.MachinesSubnet) {
-			allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), errors.New("invalid subnet ID")))
+// validateControlPlanePort validates the machines subnets and network, while enforcing proper byo subnets usage and returns a list of all validation errors.
+func validateControlPlanePort(p *openstack.Platform, n *types.Networking, ci *CloudInfo, fldPath *field.Path) (allErrs field.ErrorList) {
+	networkID := ""
+	hasIPv4Subnet := false
+	hasIPv6Subnet := false
+	if len(p.ExternalDNS) > 0 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("externalDNS"), p.ExternalDNS, "externalDNS is set, externalDNS is not supported when ControlPlanePort is set"))
+	}
```

Comment:
return allErrs here.

Reply:
The idea of not returning `allErrs` here is to allow all the validation of the `controlPlanePort` field to happen and we can return all the errors at once. The same pattern is followed by the validation of the other fields.

Let me know if you would still prefer to return it here.